src/test/java/org/alfresco/rest/DeletedNodesTest.java

```diff
@@ -230,6 +231,41 @@ public void testCreateAndRestore() throws Exception
         //Now we can restore it.
         post(URL_DELETED_NODES+"/"+document.getId()+"/restore", null, null, 200);
 
+        deleteNode(document.getId());
+
+        // Create a new nodeTargetAssoc containing target id and association type
+        NodeTargetAssoc nodeTargetAssoc = new NodeTargetAssoc();
+        nodeTargetAssoc.setTargetParentId(f1Id);
+        nodeTargetAssoc.setAssocType(ASSOC_TYPE_CM_CONTAINS);
```

Comment:
We should add a test that checks the behaviour when assocType is null.

Reply:
Thank you for catching this.